types/types.go

```diff
@@ -349,6 +349,10 @@ type SystemContext struct {
 	DockerDaemonHost string
 	// Used to skip TLS verification, off by default. To take effect DockerDaemonCertPath needs to be specified as well.
 	DockerDaemonInsecureSkipTLSVerify bool
+
+	// === dir.Transport overrides ===
+	// ShouldCompress is used to compress the blobs based on user input
+	ShouldCompress bool
```

Comment:
I never see this variable being set?

Reply:
It is set in skopeo with the value of the compress flag from skopeo copy.

Reply:
Perhaps we should name it ForceCompress.  Since if I don't set it to true, but the source is compressed, then the dest will be compressed and the variable looks like you should not be compressed.

Reply:
okay, changed to ForceCompress